packages/client/hmi-client/src/components/widgets/tera-modal.vue

```diff
@@ -12,29 +12,43 @@
  * @example
  * <modal @modal-mask-clicked="closeModal"></modal>
  */
+
+defineProps({
+	zIndex: {
+		type: String || Number,
+		default: 'var(--z-index-modal)'
+	}
+});
 </script>
 
 <template>
 	<Transition name="modal">
-		<aside @click.self="$emit('modalMaskClicked')">
-			<main>
+		<main :style="{ 'z-index': zIndex }">
+			<section>
 				<header>
-					<slot name="header"></slot>
+					<slot name="header" />
 				</header>
-				<slot></slot>
-				<section><slot name="math-editor"></slot></section>
+				<slot />
+				<section><slot name="math-editor" /></section>
 				<footer>
-					<slot name="footer"></slot>
+					<slot name="footer" />
 				</footer>
-			</main>
-		</aside>
+			</section>
+			<aside @click.self="$emit('modalMaskClicked')" />
+		</main>
 	</Transition>
 </template>
 
 <style scoped>
+main {
+	isolation: isolate;
+}
```

Comment:
This is how you do default in CSS, not by using JS, but by using custom properties

```suggestion
main {
  isolation: isolate;
  z-index: var(--z-index, var(--z-index-modal));
}
```

Reply:
added